js/utils/helpers.js

```diff
@@ -1268,16 +1271,30 @@
               }
           }
 
+          if (values && values.length > 1) {
+              conflictValues = values[1].split(',');
+          }
 
           mappedErrMessage = "The entry cannot be created/updated. ";
           if (msgTail) {
               mappedErrMessage += "Please use a different "+ msgTail +" for this record.";
           } else {
               mappedErrMessage += "Input data violates unique constraint.";
           }
-          return new module.DuplicateConflictError(errorStatusText, mappedErrMessage, generatedErrMessage);
-      }
-      else{
+
+          // parse out column names and values from generatedErrMessage
+          var conflictKeyValues = [];
+          primaryColumns.forEach(function (colName, idx) {
+              // strip off " character, if present
+              if (colName.indexOf('"') != -1) colName = colName.split('"')[1];
+              conflictKeyValues.push(colName.trim() + "=" + conflictValues[idx].trim());
+          });
+
+          var uri = reference.unfilteredReference.uri + '/' + conflictKeyValues.join('&');
+          var error = new module.DuplicateConflictError(errorStatusText, mappedErrMessage, generatedErrMessage);
+          error.uri = uri;
```

Comment:
Instead of attaching `uri` in here, it should be a property in the `DuplicateConflictError` prototype. Also it's better if you just pass the reference instead of a `uri` that chaise has to resolve again.

 So ine here

```javascript
var existingRowReference = new Reference(module.parse(uri),  reference.table.schema.catalog);
return new module.DuplicateConflictError(errorStatusText, mappedErrMessage, generatedErrMessage, existingRowReference);
```
And in the constructor of `DuplicateConflictError`, you just attach the `existingRowReference` (you most probably going to need to use a better name). So chaise can call this `exception.existingRowReference.contextualize.detailed.appLink`.